db-engine/src/main/java/com/splicemachine/db/iapi/sql/compile/CompilerContext.java

```diff
@@ -656,9 +656,9 @@
 	 */
 	boolean isReferenced(SequenceDescriptor sd);
 
-    void setDataSetProcessorType(DataSetProcessorType type);
+	void setDataSetProcessorType(DataSetProcessorType type, boolean setDSPTypeinLCC);
```

Comment:
Why is this change needed?

Reply:
I think this is an old change I made to avoid using control when we're running in the Olap Server.  But I have since fixed that in CostChoosingDataSetProcessorFactory, so I will remove the above change.

Reply:
I tried backing out this change, but it broke some Spark Explain tests. CostChoosingDataSetProcessorFactory.chooseProcessor() pulls the DataSetProcessorType from the lcc to determine whether to run on control or spark.  The lcc is not properly updated in all flows, so that is the intent of this change, to make sure that the DataSetProcessorType we've picked in the CompilerContext is propagated to the lcc.  Spark Explain needs to be executed on the Olap Server, so the lcc must always have the proper DataSetProcessorType setting.